test/unit/manipulation.js

```diff
@@ -500,6 +500,29 @@ QUnit.test( "html(String) tag-hyphenated elements (Bug #1987)", function( assert
 	assert.equal( j.children().text(), "text", "Tags with multiple hypens behave normally" );
 } );
 
+QUnit.test( "html(String) respect the spec for tag names (gh-2005) ", function ( assert ) {
+
+	assert.expect( 50 );
+
+	var wrapper, nodeName,
+		specialChars = "[ ] { } _ - = + \\ ( ) * & ^ % $ # @ ! ~ ` ' ; ?".split( " " );
+
+	specialChars.push( specialChars.join( "" ) );
+
+	function assertSupport( method, nodeName ) {
+		wrapper = jQuery( "<div></div>" );
+		wrapper[method]( "<" + nodeName + "></" + nodeName + ">" );
+		assert.ok( jQuery.nodeName( wrapper.children()[0], nodeName.toUpperCase() ), "Tagnames with special characters" );
+	}
+
+	jQuery.each( specialChars, function ( i, character ) {
+		nodeName = "valid" + character + "tagname";
+
+		assertSupport( "html", nodeName );
+		assertSupport( "append", nodeName );
```

Comment:
> (...) and negative tests for name-terminating whitespace characters.

Could you please elaborate on this request? Do you mean test support for elements with tags ending in one of the characters `\x20\t\r\n\f` i.e. `<div\t>`?